### PR TITLE
Make sponsoring(challenges & km-sponsoring) easier

### PR DESCRIPTION
**Pull request by MkuuWaUjinga**
This PR makes it possible to create a sponsoring directly on a team page.

Additionally it sets new challenges and sponsorings directly to ACCEPTED. They can be rejected by the team afterwards, if they don't want to have it.

Confirmation emails are sent but not necessary to be received.

### CODE REVIEW

File: src/main/java/backend/controller/UserController.kt [hunk 1]
```diff
@@ -282,10 +282,11 @@ class UserController(private val userService: UserService,
         sponsor.address = sponsorView.address?.toAddress() ?: sponsor.address
         sponsor.isHidden = sponsorView.isHidden ?: sponsor.isHidden
         sponsor.company = sponsorView.company ?: sponsor.company
-        sponsor.logo = sponsorView.logo?.let(::Media) ?: sponsor.logo
+        sponsor.logo = sponsorView.logo?.let(::Media) ?: null
```
Comment (nerdsupremacist) [inline src/main/java/backend/controller/UserController.kt]: This change seems pretty dangerous. If someone forgets to include the logo it gets deleted

File: src/main/resources/db/migration/V201806012220__add_emoji_support_to_posting.sql [hunk 2]
```diff
@@ -120,12 +120,12 @@ alter table posting_media convert to character set utf8mb4 collate utf8mb4_unico
 
 alter table postinglike convert to character set utf8mb4 collate utf8mb4_unicode_ci;
 
-alter table schema_version convert to character set utf8mb4 collate utf8mb4_unicode_ci;
-ALTER TABLE `schema_version` CHANGE `version` `version` VARCHAR(50)  CHARACTER SET utf8mb4  COLLATE utf8mb4_unicode_ci  NULL  DEFAULT NULL;
-ALTER TABLE `schema_version` CHANGE `description` `description` VARCHAR(200)  CHARACTER SET utf8mb4  COLLATE utf8mb4_unicode_ci  NOT NULL  DEFAULT '';
-ALTER TABLE `schema_version` CHANGE `type` `type` VARCHAR(20)  CHARACTER SET utf8mb4  COLLATE utf8mb4_unicode_ci  NOT NULL  DEFAULT '';
-ALTER TABLE `schema_version` CHANGE `script` `script` VARCHAR(1000)  CHARACTER SET utf8mb4  COLLATE utf8mb4_unicode_ci  NOT NULL  DEFAULT '';
-ALTER TABLE `schema_version` CHANGE `installed_by` `installed_by` VARCHAR(100)  CHARACTER SET utf8mb4  COLLATE utf8mb4_unicode_ci  NOT NULL  DEFAULT '';
+-- alter table schema_version convert to character set utf8mb4 collate utf8mb4_unicode_ci;
```
Comment (nerdsupremacist) [inline src/main/resources/db/migration/V201806012220__add_emoji_support_to_posting.sql]: You shouldn't have committed that

File: src/main/java/backend/controller/UserController.kt [hunk 3]
```diff
@@ -282,10 +282,11 @@ class UserController(private val userService: UserService,
         sponsor.address = sponsorView.address?.toAddress() ?: sponsor.address
         sponsor.isHidden = sponsorView.isHidden ?: sponsor.isHidden
         sponsor.company = sponsorView.company ?: sponsor.company
-        sponsor.logo = sponsorView.logo?.let(::Media) ?: sponsor.logo
+        sponsor.logo = sponsorView.logo?.let(::Media) ?: null
 
         val urlString = sponsorView.url
         if (urlString != null) sponsor.url = Url(urlString)
+        else sponsor.url = null
```
Comment (nerdsupremacist) [inline src/main/java/backend/controller/UserController.kt]: Same here. If you don't set it it will be deleted

**Review by nerdsupremacist — CHANGES_REQUESTED** [review]

File: src/main/java/backend/controller/UserController.kt (hunk 1 above)
Comment (nerdsupremacist) [reply src/main/java/backend/controller/UserController.kt]: Also you don't need the Elvis operator in that case

File: src/main/resources/db/migration/V201806012220__add_emoji_support_to_posting.sql (hunk 2 above)
Comment (dominikglandorf) [reply src/main/resources/db/migration/V201806012220__add_emoji_support_to_posting.sql]: build is still failing for me, without uncommenting this

**Review by nerdsupremacist — APPROVED** [review]

**Pull request merged**